ego/cli/install.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Edgeless Systems GmbH.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package cli
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"strings"
+
+	"github.com/klauspost/cpuid/v2"
+	"github.com/zcalusic/sysinfo"
+)
+
+const shellToUse = "bash"
+
+var ErrOSNotSupported = errors.New("os not supported")
+var ErrTargetNotSupported = errors.New("install target not supported")
+var ErrInstallationFail = errors.New("installation fail")
+var ErrExitCodeValue = errors.New("exit code not 0")
+var ErrSysInfoFail = errors.New("could not determine os details")
+
+type installInfoV1 struct {
+	Desc     map[string]string
+	Commands map[string]map[string][]string
+}
+
+type installInfo struct {
+	V1 installInfoV1 `json:"1.0"`
+}
+
+func (i installInfo) getDescriptions() map[string]string {
+	return i.V1.Desc
+}
+
+func (i installInfo) getCommands(osDetails string) map[string][]string {
+	return i.V1.Commands[osDetails]
+}
+
+// Determine relevant information about os and sgx compatibility. Then pass them with osDetails to install
+func (c *Cli) Install(ask func(string) bool, component string) error {
+
+	var si sysinfo.SysInfo
+	si.GetSysInfo()
```

Comment:
si.GetSysInfo breaks os.Stdin on my system. The following minimal program
```go
func main() {
	var si sysinfo.SysInfo
	si.GetSysInfo()
	fmt.Println(bufio.NewReader(os.Stdin).ReadString('\n'))
}
```
prints ` read /dev/stdin: bad file descriptor`. So ego install will fail on my system when ask() is called. Does it work on yours?

Reply:
Would it not make sense to use another package (or DIY) distro detection here? The `GetSysInfo()` extension does a lot of other checks to get machine information, which we do not need at all for an OS detection. 

It's quite unfourtunate that they have a function which specifically detects and fills the OS information, but is marked as private :/

Reply:
Yes, the same error occured on my system. I copied the necessary parts from the script to extract the vendor and the version and it works now.